tests/poller_scaling_test.go

```diff
@@ -0,0 +1,207 @@
+	sdkClient sdkclient.Client
+}
+
+func (s *PollerScalingIntegSuite) mustToPayload(v any) *commonpb.Payload {
+	conv := converter.GetDefaultDataConverter()
+	payload, err := conv.ToPayload(v)
+	s.NoError(err)
+	return payload
+}
+
+func TestPollerScalingFunctionalSuite(t *testing.T) {
+	t.Parallel()
+	suite.Run(t, new(PollerScalingIntegSuite))
+}
+
+func (s *PollerScalingIntegSuite) SetupSuite() {
+	dynamicConfigOverrides := map[dynamicconfig.Key]any{
+		// Force one partition so we can reliably see the backlog
+		dynamicconfig.MatchingNumTaskqueueReadPartitions.Key():     1,
+		dynamicconfig.MatchingNumTaskqueueWritePartitions.Key():    1,
+		dynamicconfig.MatchingPollerScalingBacklogAgeScaleUp.Key(): 50 * time.Millisecond,
+	}
+	s.FunctionalTestBase.SetupSuiteWithDefaultCluster(testcore.WithDynamicConfigOverrides(dynamicConfigOverrides))
+}
+
+func (s *PollerScalingIntegSuite) SetupTest() {
+	s.FunctionalTestSuite.SetupTest()
+
+	var err error
+	s.sdkClient, err = sdkclient.Dial(sdkclient.Options{
+		HostPort:  s.FrontendGRPCAddress(),
+		Namespace: s.Namespace().String(),
+	})
+	s.NoError(err)
+}
+
+func (s *PollerScalingIntegSuite) TearDownTest() {
+	if s.sdkClient != nil {
+		s.sdkClient.Close()
+	}
+}
+
+func (s *PollerScalingIntegSuite) TestPollerScalingSimpleBacklog() {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	tq := testcore.RandomizeStr(s.T().Name())
+	endpointName := testcore.RandomizedNexusEndpoint(s.T().Name())
+	listenAddr := nexustest.AllocListenAddress()
+
+	_, err := s.OperatorClient().CreateNexusEndpoint(ctx, &operatorservice.CreateNexusEndpointRequest{
+		Spec: &nexuspb.EndpointSpec{
+			Name: endpointName,
+			Target: &nexuspb.EndpointTarget{
+				Variant: &nexuspb.EndpointTarget_External_{
+					External: &nexuspb.EndpointTarget_External{
+						Url: "http://" + listenAddr,
+					},
+				},
+			},
+		},
+	})
+	s.NoError(err)
+
+	// Queue up a couple workflows
+	runHandles := make([]sdkclient.WorkflowRun, 0)
+	for i := 0; i < 5; i++ {
+		run, err := s.sdkClient.ExecuteWorkflow(
+			ctx, sdkclient.StartWorkflowOptions{TaskQueue: tq}, "wf")
+		s.NoError(err)
+		runHandles = append(runHandles, run)
+	}
+
+	// Wait for the backlog age to pass the config threshold
+	//time.Sleep(5000 * time.Millisecond)
+
+	// Poll for a task and see attached decision is to scale up b/c of backlog
+	feClient := s.FrontendClient()
+	// This needs to be done in an eventually loop because nexus endpoints don't become available immediately...
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		resp, err := feClient.PollWorkflowTaskQueue(ctx, &workflowservice.PollWorkflowTaskQueueRequest{
+			Namespace: s.Namespace().String(),
+			TaskQueue: &taskqueuepb.TaskQueue{Name: tq, Kind: enumspb.TASK_QUEUE_KIND_NORMAL},
+		})
+		assert.NoError(t, err)
+		assert.NotNil(t, resp.PollerScalingDecision)
+		assert.GreaterOrEqual(t, int32(1), resp.PollerScalingDecision.PollRequestDeltaSuggestion)
```

Comment:
`assert` calls don't stop execution and you can't use `require` in `EventuallyWithT` blocks. You'll want to ensure your code won't panic if it gets passed an assertion.

Reply:
Yeah, I wasn't expecting them to, if the eventually passes that's all I need here

Reply:
If the test panics it will fail without rerunning the eventually block. I don't think that's what you want.

Reply:
I'm not following you. This is doing exactly what I expect, it re-runs the block until the asserts pass. Nothing is panicking.

Reply:
I think what @bergundy is saying is that line 138 will run and panic even when `assert.NotNil(t, resp.PollerScalingDecision)` is false. You may want to say something like:
```suggestion
		assert.NoError(t, err)
		if assert.NotNil(t, resp.PollerScalingDecision) {
		  assert.GreaterOrEqual(t, int32(1), resp.PollerScalingDecision.PollRequestDeltaSuggestion)
```